### PR TITLE
Add autograd recipe

### PR DESCRIPTION
**Pull request by richardotis**
cc @jakirkham

### CODE REVIEW

File: recipes/autograd/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  # Python imports
+  imports:
+    - autograd
+    - autograd.numpy
+    - autograd.scipy
+    - autograd.scipy.stats
+
+about:
+  home: https://github.com/HIPS/autograd
+  license: MIT License
```
Comment (jakirkham) [inline recipes/autograd/meta.yaml]: Please shorten to `MIT`.

File: recipes/autograd/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Note: there are many handy hints in commenets in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
```
Comment (jakirkham) [inline recipes/autograd/meta.yaml]: There are a bunch of explanatory comments from the example. Please remove them.

**Comment by jakirkham** [comment]
@ericmjl, would you like to help review this given your interest in `autograd`?

File: recipes/autograd/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+{% set sha256 = "ef94df9ea2c61ff0785b1726096e5c424ceaea60f93a238d61a16d16097c3a48" %}
+
+package:
+  name: autograd
+  version: {{ version }}
+
+source:
+  fn: autograd-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/s/autograd/autograd-{{ version }}.tar.gz
+  # md5 is OK but sha256 is desirable.
```
Comment (jakirkham) [inline recipes/autograd/meta.yaml]: Please remove this comment too.

File: recipes/autograd/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - scipy >=0.17
+
+  run:
+    - python
+    - numpy >=1.10
+    - future >=0.15.2
+    - scipy >=0.17
+
+test:
+  # Python imports
```
Comment (jakirkham) [inline recipes/autograd/meta.yaml]: Also this one.

File: recipes/autograd/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    - autograd.scipy.stats
+
+about:
+  home: https://github.com/HIPS/autograd
+  license: MIT
+  summary: 'Efficiently computes derivatives of numpy code.'
+
+extra:
+  recipe-maintainers:
+    - richardotis
```
Comment (jakirkham) [inline recipes/autograd/meta.yaml]: nit: please add terminal newline.

**Comment by jakirkham** [comment]
Couple minor things left. Otherwise LGTM. Will wait to see if @ericmjl would like to add anything else.

File: recipes/autograd/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,45 @@
+{% set version = "1.1.6" %}
+{% set sha256 = "ef94df9ea2c61ff0785b1726096e5c424ceaea60f93a238d61a16d16097c3a48" %}
+
+package:
+  name: autograd
+  version: {{ version }}
+
+source:
+  fn: autograd-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/s/autograd/autograd-{{ version }}.tar.gz
```
Comment (jakirkham) [inline recipes/autograd/meta.yaml]: `/s/` -> `/a/`

**Comment by jakirkham** [comment]
Seems it is failing to download due to a minor issue.

**Comment by richardotis** [comment]
Thanks for your patience @jakirkham -- I wasn't very familiar with how the new pypi URLs work.

**Comment by jakirkham** [comment]
No worries. Yeah, that `s` surprised me before. It is easy to think it means `source` or similar. Instead it is just the first letter of the package name. Might try to update the example to make this more clear.

File: recipes/autograd/meta.yaml (hunk 5 above)
Comment (ericmjl) [inline recipes/autograd/meta.yaml]: Please add me as a recipe maintainer! :smile:

**Comment by ericmjl** [comment]
If all tests pass, the @jakirkham, looks good to me as well. Only final request is to add me as a recipe maintainer.

**Comment by jakirkham** [comment]
Thanks everyone.

**Pull request merged**